scripts/sanitycheck

```diff
@@ -316,6 +317,86 @@ class Handler:
         self.lock.release()
         return ret
 
+
+class DeviceHandler(Handler):
+
+    def __init__(self, instance):
+        """Constructor
+
+        @param instance Test Instance
+        """
+        super().__init__(instance)
+
+        self.instance = instance
+        self.timeout = instance.test.timeout
+        self.sourcedir = instance.test.code_location
+        self.outdir = instance.outdir
+        self.run_log = os.path.join(self.outdir, "run.log")
+        self.handler_log = os.path.join(self.outdir, "handler.log")
+        self.returncode = 0
+        self.set_state("running", {})
+
+    def monitor_serial(self, ser, harness):
+        log_out_fp = open(self.handler_log, "wt")
+
+        while ser.isOpen():
+            serial_line = ser.readline().decode('utf-8', 'ignore')
+            if serial_line:
+                verbose("DEVICE: {0}".format(serial_line.rstrip()))
+            log_out_fp.write(serial_line)
+            log_out_fp.flush()
+            harness.handle(serial_line.rstrip())
+            if harness.state:
+                ser.close()
+                break
+
+        log_out_fp.close()
+
+    def handle(self):
+        out_state = "failed"
+
+        device = options.device_serial
+        ser = serial.Serial(
+                device,
+                baudrate=115200,
+                parity=serial.PARITY_NONE,
+                stopbits=serial.STOPBITS_ONE,
+                bytesize=serial.EIGHTBITS,
+                timeout=self.timeout
+                )
+
+        ser.flush()
+
+        harness_name = self.instance.test.harness.capitalize()
+        harness_import = HarnessImporter(harness_name)
+        harness = harness_import.instance
+        harness.configure(self.instance)
+
+        t = threading.Thread(target=self.monitor_serial, args=(ser, harness))
+        t.start()
+
+        if options.make:
+            generator_cmd = "$(MAKE)"
+        else:
+            generator_cmd = "ninja"
+
+        command = [generator_cmd, "-C", self.outdir, "flash"]
+
+        subprocess.check_output(command, stderr=subprocess.PIPE)
+
+        t.join(self.timeout)
+        if t.is_alive():
+            out_state = "timeout"
+            ser.close()
+
+        if ser.isOpen():
+            ser.close()
+
+        if harness.state:
+            self.set_state(harness.state, {})
```

Comment:
Is this parsing the serial output for pass/fail?

Reply:
yes, in scripts/sanity_chk/harness.py